pkg/arvo/app/chat-hook.hoon

```diff
@@ -0,0 +1,364 @@
+::  chat-hook:
+::  mirror chat data from foreign to local based on read permissions
+::  allow sending chat messages to foreign paths based on write perms
+::
+/-  *permission-store, *chat-hook
+/+  *chat-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%chat-update chat-update]]
+      [%quit ~]
+      [%poke wire dock poke]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  synced=(map path ship)
+      boned=(map wire (list bone))
+  ==
+::
++$  poke
+  $%  [%chat-action chat-action]
+      [%permission-action permission-action]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    :_  this
+    [ost.bol %peer /permissions [our.bol %permission-store] /updates]~
+  [~ this(+<+ u.old)]
+::
+++  poke-noun
+  |=  a=*
+  ^-  (quip move _this)
+  ~&  synced
+  ~&  boned
+  [~ this]
+::
+++  poke-json
+  |=  jon=json
+  ^-  (quip move _this)
+  (poke-chat-action (json-to-action jon))
+::
+++  poke-chat-action
+  |=  act=chat-action
+  ^-  (quip move _this)
+  ?>  ?=(%message -.act)
+  ::  local
+  :_  this
```

Comment:
It's strange to produce the core first when there are state updates below. I guess it works because `+this` is an arm and it's evaluation is deferred, but I don't think it's a good pattern -- it'd be better to put it after the state updates and move-construction at each "return" point. (I have no objections to the use of `:_` in such places)

Reply:
What state updates? The =. statements apply to arguments to the gate that are then passed in a move.

Reply:
Ah, I just saw the `=.` ... my bad.